### DISCUSSION TITLE
Fine Tune OSPF Underlay

### DISCUSSION THREAD

**Discussion by theotherguy2175** [aristanetworks/avd#3115] [Q&A]
We are trying to replicate our current config into something AVD can create. This is our current ospf config:

router ospf 1
   router-id 10.x.x.x
   network 10.x.x.x/32 area 0.0.0.0
   ....
   network 10.x.x.x/32 area 0.0.0.0
   max-lsa 12000
   
 However AVD by default will  add just the interfaces:
 
router ospf 1
   router-id 10.x.x.x
   passive-interface default
   no passive-interface Ethernet49
   no passive-interface Ethernet50
   no passive-interface Vlan4093
   no passive-interface Ethernet7
   max-lsa 120000
   
Is there a way to modify AVD configs to more closely resemble the original config?

**Comment by ClausHolbechArista** [comment]
Hi @theotherguy2175, thank you for your question.

The AVD `eos_designs` role attempts to create config based on what we consider the "best practice", where areas are set on the interfaces instead of under `router ospf` and using `passive-interface default`.

It is possible to override the generated configuration using `structured_config` or `custom_structured_configuration_router_ospf`, but it will not be very easy to work with.

As a last resort we also have `raw_eos_cli` knobs in various areas of the data models.
